--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@
 
 
 # CMake minimum version
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 # Sub directories
 add_subdirectory(Source/DNSPing)
--- a/Source/DNSPing/CMakeLists.txt
+++ b/Source/DNSPing/CMakeLists.txt
@@ -18,7 +18,7 @@
 
 
 # CMake minimum version
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 # Project name, build type and list all source files.
 project(DNSPing)
--- a/Source/FileHash/CMakeLists.txt
+++ b/Source/FileHash/CMakeLists.txt
@@ -18,7 +18,7 @@
 
 
 # CMake minimum version
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 # Project name, build type, list all source files and sub directories.
 project(FileHash)
