Index: mtproxy/common/server-functions.c
===================================================================
--- mtproxy.orig/common/server-functions.c
+++ mtproxy/common/server-functions.c
@@ -35,7 +35,6 @@
 #include <arpa/inet.h>
 #include <assert.h>
 #include <errno.h>
-#include <execinfo.h>
 #include <fcntl.h>
 #include <getopt.h>
 #include <grp.h>
@@ -168,10 +167,8 @@ const char *get_version_string (void) {
 }
 
 void print_backtrace (void) {
-  void *buffer[64];
-  int nptrs = backtrace (buffer, 64);
   kwrite (2, "\n------- Stack Backtrace -------\n", 33);
-  backtrace_symbols_fd (buffer, nptrs, 2);
+  kwrite (2, "Stack trace unavailable in musl\n", 24);
   kwrite (2, "-------------------------------\n", 32);
   const char *s = get_version_string ();
   if (s) {
