--- a/ddns-scripts/files/etc/init.d/ddns
+++ b/ddns-scripts/files/etc/init.d/ddns
@@ -2,10 +2,6 @@
 START=95
 STOP=10
 
-boot() {
-	return 0
-}
-
 reload() {
 	/usr/lib/ddns/dynamic_dns_updater.sh -- reload
 	return 0

--- a/luci-app-ddns/Makefile
+++ b/luci-app-ddns/Makefile
@@ -12,7 +12,7 @@ PKG_LICENSE:=Apache-2.0
 PKG_MAINTAINER:=Ansuel Smith <ansuelsmth@gmail.com>
 
 LUCI_TITLE:=LuCI Support for Dynamic DNS Client (ddns-scripts)
-LUCI_DEPENDS:=+luci-base +ddns-scripts
+LUCI_DEPENDS:=+luci-base +ddns-scripts +ddns-scripts-aliyun +ddns-scripts-dnspod-v3 +ddns-scripts-cloudflare
 
 include ../../luci.mk
 

--- a/ddns-scripts/files/etc/init.d/ddns
+++ b/ddns-scripts/files/etc/init.d/ddns
@@ -165,6 +165,15 @@ define Package/ddns-scripts-dnspod-v3/description
   Dynamic DNS Client scripts extension for dnspod.tencentcloudapi.com API v3 (require curl)
 endef
 
+define Package/ddns-scripts-aliyun
+  $(call Package/ddns-scripts/Default)
+  TITLE:=Extension for aliyun.com API
+  DEPENDS:=ddns-scripts +curl +bind-host +openssl-util
+endef
+
+define Package/ddns-scripts-aliyun/description
+  Dynamic DNS Client scripts extension for aliyun.com API (require curl)
+endef
 
 define Package/ddns-scripts-noip
   $(call Package/ddns-scripts/Default)
@@ -401,6 +410,7 @@ define Package/ddns-scripts-services/install
 	rm $(1)/usr/share/ddns/default/digitalocean.com-v2.json
 	rm $(1)/usr/share/ddns/default/dnspod.cn.json
 	rm $(1)/usr/share/ddns/default/dnspod.cn-v3.json
+	rm $(1)/usr/share/ddns/default/aliyun.com.json
 	rm $(1)/usr/share/ddns/default/no-ip.com.json
 	rm $(1)/usr/share/ddns/default/bind-nsupdate.json
 	rm $(1)/usr/share/ddns/default/route53-v1.json
@@ -575,6 +585,23 @@ fi
 exit 0
 endef
 
+define Package/ddns-scripts-aliyun/install
+	$(INSTALL_DIR) $(1)/usr/lib/ddns
+	$(INSTALL_BIN) ./files/usr/lib/ddns/update_aliyun_com.sh \
+		$(1)/usr/lib/ddns
+
+	$(INSTALL_DIR) $(1)/usr/share/ddns/default
+	$(INSTALL_DATA) ./files/usr/share/ddns/default/aliyun.com.json \
+		$(1)/usr/share/ddns/default/
+endef
+
+define Package/ddns-scripts-aliyun/prerm
+#!/bin/sh
+if [ -z "$${IPKG_INSTROOT}" ]; then
+	/etc/init.d/ddns stop
+fi
+exit 0
+endef
 
 define Package/ddns-scripts-noip/install
 	$(INSTALL_DIR) $(1)/usr/lib/ddns
@@ -796,6 +823,7 @@ $(eval $(call BuildPackage,ddns-scripts-godaddy))
 $(eval $(call BuildPackage,ddns-scripts-digitalocean))
 $(eval $(call BuildPackage,ddns-scripts-dnspod))
 $(eval $(call BuildPackage,ddns-scripts-dnspod-v3))
+$(eval $(call BuildPackage,ddns-scripts-aliyun))
 $(eval $(call BuildPackage,ddns-scripts-noip))
 $(eval $(call BuildPackage,ddns-scripts-nsupdate))
 $(eval $(call BuildPackage,ddns-scripts-route53))
