--- a/luci-app-aria2/Makefile
+++ b/luci-app-aria2/Makefile
@@ -7,7 +7,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=LuCI Support for Aria2
-LUCI_DEPENDS:=+luci-base +aria2
+LUCI_DEPENDS:=+luci-base +aria2 +ariang
 
 PKG_MAINTAINER:=Xingwang Liao <kuoruan@gmail.com>
 PKG_LICENSE:=MIT

--- a/luci-app-samba4/Makefile
+++ b//luci-app-samba4/Makefile
@@ -3,7 +3,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=Network Shares - Samba 4 SMB/CIFS fileserver
-LUCI_DEPENDS:=+luci-base +samba4-server
+LUCI_DEPENDS:=+luci-base +samba4-server +autoshare-samba
 
 PKG_LICENSE:=Apache-2.0
 PKG_MAINTAINER:=Andy Walsh <andy.walsh44+github@gmail.com>

--- a/luci-app-ksmbd/Makefile
+++ b/luci-app-ksmbd/Makefile
@@ -3,7 +3,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=Network Shares - Ksmbd the SMB kernel fileserver
-LUCI_DEPENDS:=+luci-base +ksmbd-server
+LUCI_DEPENDS:=+luci-base +ksmbd-server +autoshare-ksmbd
 
 PKG_LICENSE:=Apache-2.0
 PKG_MAINTAINER:=Richard Yu <yurichard3839@gmail.com>

--- a/luci-app-ua2f/Makefile
+++ b/luci-app-ua2f/Makefile
@@ -2,7 +2,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=ua2f for luci
-LUCI_DEPENDS:=
+LUCI_DEPENDS:=+ua2f
 LUCI_PKGARCH:=all
 PKG_VERSION:=5.2
 PKG_RELEASE:=5

--- a/k3screenctrl/Makefile
+++ b/k3screenctrl/Makefile
@@ -19,7 +19,7 @@ TARGET_CFLAGS+= -D_GNU_SOURCE
 define Package/k3screenctrl
   SECTION:=utils
   CATEGORY:=Utilities
-  DEPENDS:=@(TARGET_bcm53xx_generic_DEVICE_phicomm_k3||TARGET_bcm53xx_generic_DEVICE_phicomm-k3||TARGET_bcm53xx_DEVICE_phicomm-k3) +@KERNEL_DEVMEM +coreutils +coreutils-od +bash +curl
+  DEPENDS:=@(TARGET_bcm53xx_generic_DEVICE_phicomm_k3||TARGET_bcm53xx_generic_DEVICE_phicomm-k3||TARGET_bcm53xx_DEVICE_phicomm-k3||TARGET_DEVICE_bcm53xx_generic_DEVICE_phicomm_k3) +@KERNEL_DEVMEM +coreutils +coreutils-od +bash +curl
   TITLE:=LCD screen controller on PHICOMM K3
   URL:=https://github.com/lwz322/k3screenctrl.git
 endef

--- a/fullconenat/Makefile
+++ b/fullconenat/Makefile
@@ -36,7 +36,7 @@ define Package/ip6tables-mod-fullconenat
   SECTION:=net
   CATEGORY:=Network
   TITLE:=FULLCONENAT ip6tables extension
-  DEPENDS:=ip6tables +kmod-nf-nat6 +kmod-ipt-fullconenat +ip6tables-mod-nat
+  DEPENDS:=+ip6tables +kmod-nf-nat6 +kmod-ipt-fullconenat +ip6tables-mod-nat
 endef
 
 define KernelPackage/ipt-fullconenat

--- a/phtunnel/Makefile
+++ b/phtunnel/Makefile
@@ -36,7 +36,6 @@ PKG_SOURCE:=phtunnel
 PKG_SOURCE_URL:=https://mirrors.oray.com/orayos/packages/$(PKG_NAME)/$(PKG_ARCH_PHT)/$(PKG_VERSION)/bin
 PKG_HASH:=skip
 
-PKG_FLAGS:=nonshared
 PKG_MAINTAINER:=Oray <developer@oray.com>
 
 include $(INCLUDE_DIR)/package.mk

--- a/pgyvpn/Makefile
+++ b/pgyvpn/Makefile
@@ -36,7 +36,6 @@ PKG_SOURCE:=pgyvpnsvr
 PKG_SOURCE_URL:=https://mirrors.oray.com/orayos/packages/$(PKG_NAME)/$(PKG_ARCH_PGYYPN)/$(PKG_VERSION)/bin
 PKG_HASH:=skip
 
-PKG_FLAGS:=nonshared
 PKG_MAINTAINER:=Oray <developer@oray.com>
 
 include $(INCLUDE_DIR)/package.mk

--- a/wifidog-wiwiz/Makefile
+++ b/wifidog-wiwiz/Makefile
@@ -21,7 +21,7 @@ define Package/wifidog-wiwiz
   SUBMENU:=Portal
   SECTION:=net
   CATEGORY:=Wiwiz/PinPinWiFi
-  DEPENDS:=+iptables-mod-extra +iptables-mod-ipopt +iptables-mod-nat-extra +libpthread +curl
+  DEPENDS:=+luci-app-eqos +dcc2-wiwiz-nossl +iptables-mod-extra +iptables-mod-ipopt +iptables-mod-nat-extra +libpthread +curl
   TITLE:=wifidog-wiwiz
   URL:=http://www.wiwiz.com
   MENU:=1

--- a/luci-app-rclone/Makefile
+++ b/luci-app-rclone/Makefile
@@ -9,7 +9,7 @@ PKG_VERSION:=1.4.1-3
 PKG_RELEASE:=
 
 LUCI_TITLE:=LuCI support for Rclone
-LUCI_DEPENDS:=+rclone-config \
+LUCI_DEPENDS:=+rclone-config +fuse-overlayfs +fuse3-utils \
               +PACKAGE_$(PKG_NAME)_INCLUDE_rclone-webui:rclone-webui-react \
               +PACKAGE_$(PKG_NAME)_INCLUDE_rclone-ng:rclone-ng
 LUCI_PKGARCH:=all

--- a/luci-app-advancedplus/Makefile
+++ b/luci-app-advancedplus/Makefile
@@ -7,7 +7,7 @@ include $(TOPDIR)/rules.mk
 NAME:=advancedplus
 PKG_NAME:=luci-app-$(NAME)
 LUCI_TITLE:=LuCI support for Kucat theme setting by sirpdboy
-LUCI_DEPENDS:=+luci-compat +curl +wget +libustream-openssl
+LUCI_DEPENDS:=+luci-compat +curl
 LUCI_PKGARCH:=all
 
 PKG_VERSION:=1.8.0

--- a/luci-app-einat/Makefile
+++ b/luci-app-einat/Makefile
@@ -9,7 +9,7 @@ LUCI_NAME:=luci-app-einat
 
 LUCI_TITLE:=LuCI Support for einat
 LUCI_PKGARCH:=all
-LUCI_DEPENDS:=
+LUCI_DEPENDS:=+einat-ebpf
 
 LUCI_DESCRIPTION:=eBPF-based Endpoint-Independent NAT
