--- a/luci-app-wifidog/root/etc/init.d/wifidog
+++ b/luci-app-wifidog/root/etc/init.d/wifidog
@@ -150,3 +150,8 @@ start_service() {
 status() {
     /usr/bin/wdctl status
 }
+
+service_triggers() {
+	procd_add_reload_trigger "wifidog"
+}
+

--- a/luci-app-docker/root/etc/init.d/docker
+++ b/luci-app-docker/root/etc/init.d/docker
@@ -19,3 +19,7 @@ start_service() {
 	procd_set_param limits nofile="${nofile} ${nofile}"
 	procd_close_instance
 }
+
+service_triggers() {
+	procd_add_reload_trigger "dockerd"
+}

--- a/luci-app-leigod-acc/root/etc/init.d/acc
+++ b/luci-app-leigod-acc/root/etc/init.d/acc
@@ -164,3 +164,6 @@ start_service() {
     start_service_daemon
 }
 
+service_triggers() {
+      procd_add_reload_trigger "acc"
+}

--- a/luci-app-gpoint/root/etc/init.d/gpoint
+++ b/luci-app-gpoint/root/etc/init.d/gpoint
@@ -30,3 +30,7 @@ stop_service() {
 	echo "gpoint not running"
    fi
 }
+
+service_triggers() {
+      procd_add_reload_trigger "gpoint"
+}

--- a/luci-app-xupnpd/root/etc/init.d/xupnpd
+++ b/luci-app-xupnpd/root/etc/init.d/xupnpd
@@ -19,3 +19,7 @@ start_service() {
 	procd_set_param respawn
 	procd_close_instance
 }
+
+service_triggers() {
+      procd_add_reload_trigger "xupnpd"
+}

--- a/luci-app-packagesync/root/etc/init.d/packagesync
+++ b/luci-app-packagesync/root/etc/init.d/packagesync
@@ -290,3 +290,7 @@ boot() {
 	mkdir -p "$RESULTPATH" 2>/dev/null
     return 0
 }
+
+service_triggers() {
+      procd_add_reload_trigger "packagesync"
+}

--- a/luci-app-change-mac/root/etc/init.d/change-mac
+++ b/luci-app-change-mac/root/etc/init.d/change-mac
@@ -101,3 +101,7 @@ restore() {
 restart() {
 	start
 }
+
+service_triggers() {
+      procd_add_reload_trigger "change-mac"
+}

--- a/mmdvm-luci/luci-app-mmdvm/root/etc/init.d/dmrid
+++ b/mmdvm-luci/luci-app-mmdvm/root/etc/init.d/dmrid
@@ -41,4 +41,8 @@ status() {
     fi
 
     echo "${APP}:		[${STATUS}]"
-}
\ No newline at end of file
+}
+
+service_triggers() {
+      procd_add_reload_trigger "dmrid"
+}

--- a/pgyvpn/files/etc/init.d/pgyvpn
+++ b/pgyvpn/files/etc/init.d/pgyvpn
@@ -105,6 +105,10 @@ start_service()
 
 }
 
+service_triggers() {
+	procd_add_reload_trigger "pgyvpn"
+}
+
 stop_service()
 {
 	#remove p2p firewall

--- a/phtunnel/files/etc/init.d/phtunnel
+++ b/phtunnel/files/etc/init.d/phtunnel
@@ -59,6 +59,10 @@ start_service()
 	procd_close_instance
 }
 
+service_triggers() {
+	procd_add_reload_trigger "phtunnel"
+}
+
 stop_service()
 {
 	echo 'now stop phtunnel'

--- a/luci-app-sms-tool/root/etc/init.d/smsled
+++ b/luci-app-sms-tool/root/etc/init.d/smsled
@@ -20,6 +20,10 @@ stop_service() {
 	return 0
 }
 
+service_triggers() {
+	procd_add_reload_trigger "sms_tool"
+}
+
 restart_service() {
     stop_service
     start_service
