--- a/frp/Makefile
+++ b/frp/Makefile
@@ -27,15 +27,6 @@ define Package/frp/install
 	$(INSTALL_BIN) $(GO_PKG_BUILD_BIN_DIR)/$(2) $(1)/usr/bin/
 	$(INSTALL_DIR) $(1)/etc/frp/$(2).d/
 	$(INSTALL_DATA) $(PKG_BUILD_DIR)/conf/$(2)_full.ini $(1)/etc/frp/$(2).d/
-	$(INSTALL_DIR) $(1)/etc/config/
-	$(INSTALL_CONF) ./files/$(2).config $(1)/etc/config/$(2)
-	$(INSTALL_DIR) $(1)/etc/init.d/
-	$(INSTALL_BIN) ./files/$(2).init $(1)/etc/init.d/$(2)
-
-	if [ -r ./files/$(2).uci-defaults ]; then \
-		$(INSTALL_DIR) $(1)/etc/uci-defaults; \
-		$(INSTALL_DATA) ./files/$(2).uci-defaults $(1)/etc/uci-defaults/$(2); \
-	fi
 endef
 
 define Package/frp/template
@@ -53,10 +44,6 @@ define Package/frp/template
     a NAT or firewall to the internet.
   endef
 
-  define Package/$(1)/conffiles
-/etc/config/$(1)
-  endef
-
   define Package/$(1)/install
     $(call Package/frp/install,$$(1),$(1))
   endef
