--- a/homebox/Makefile
+++ b/homebox/Makefile
@@ -52,8 +52,11 @@ define Build/Compile
 endef
 
 define Package/$(PKG_NAME)/install
-	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_DIR) $(1)/usr/bin $(1)/etc/config $(1)/etc/init.d $(1)/etc/uci-defaults
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/server-linux-$(H_ARCH) $(1)/usr/bin/homebox
+	$(INSTALL_CONF) ./files/homebox.config $(1)/etc/config/homebox
+	$(INSTALL_BIN) ./files/homebox.init $(1)/etc/init.d/homebox
+	$(INSTALL_BIN) ./files/homebox.uci-default $(1)/etc/uci-defaults/homebox
 endef
 
 $(eval $(call BuildPackage,$(PKG_NAME)))
