--- a/luci-app-kcptun/htdocs/luci-static/resources/view/kcptun.js
+++ b/luci-app-kcptun/htdocs/luci-static/resources/view/kcptun.js
@@ -4,8 +4,6 @@
 'require form';
 'require rpc';
 'require uci';
-'require tools.widgets as widgets';
-'require tools.github as github';
 
 var callServiceList = rpc.declare({
 	object: 'service',
@@ -50,7 +48,7 @@ return view.extend({
 		var m, s, o;
 
 		m = new form.Map('kcptun', _('kcptun'));
-		m.description = github.luci_desc('kcptun is a Stable & Secure Tunnel Based On KCP with N:M Multiplexing.', 'liudf0716', 'kcptun');
+		m.description = _("kcptun is a Stable & Secure Tunnel Based On KCP with N:M Multiplexing.");
 
 		// add kcptun-client status section and option 
 		s = m.section(form.NamedSection, '_status');

--- a/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js
+++ b/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js
@@ -6,7 +6,6 @@
 'require uci';
 'require fs';
 'require tools.widgets as widgets';
-'require tools.github as github';
 'require tools.firewall as fwtool';
 
 var callServiceList = rpc.declare({
@@ -44,7 +43,7 @@ return view.extend({
 		var m, s, o, ss;
 
 		m = new form.Map('wifidogx', _('ApFree-WiFiDog'));
-		m.description = github.desc('apfree-wifidog offers a stable and secure captive portal solution.', 'liudf0716', 'apfree-wifidog');
+		m.description = _("apfree-wifidog is a Stable & Secure captive portal solution.");
 		
 
 		s = m.section(form.NamedSection, 'common',  _('Configuration'));

--- a/luci-app-yt-dlp/htdocs/luci-static/resources/view/yt-dlp.js
+++ b/luci-app-yt-dlp/htdocs/luci-static/resources/view/yt-dlp.js
@@ -6,7 +6,6 @@
 'require uci';
 'require network';
 'require rpc';
-'require tools.github as github';
 
 return view.extend({
     load: function() {
@@ -121,7 +120,7 @@ return view.extend({
 				E('th', { 'class': 'th', 'style': 'width:20%' }, _('Action')),
 			])
 		]);
-        var description = github.luci_desc('Download video from Youtube, Facebook, Twitter, etc.', 'liudf0716', 'yt-dlp')
+        var description = _("Download video from Youtube, Facebook, Twitter, etc.");
         
         var v = E([], [
             E('h2', {}, [ _('Download Video') ]),
