--- a/luci-app-upnp/root/usr/share/rpcd/ucode/luci.upnp
+++ b/luci-app-upnp/root/usr/share/rpcd/ucode/luci.upnp
@@ -46,7 +46,7 @@ const methods = {
 
 			if (ipt) {
 				for (let line = ipt.read('line'); length(line); line = ipt.read('line')) {
-					let m = match(line, /^([0-9]+)\s+([a-z]+).+dpt:([0-9]+) to:(\S+):([0-9]+)/);
+					let m = match(line, /^([0-9]+).*\s+([a-z]+).+dpt:([0-9]+) to:(\S+):([0-9]+)/);
 
 					if (m) {
 						push(rules, {

--- a/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js
+++ b/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js
@@ -192,6 +192,9 @@ return view.extend({
 		o = s.taboption('advanced', form.Value, 'upnp_lease_file', _('Service lease file'));
 		o.placeholder = '/var/run/miniupnpd.leases';
 
+		s.taboption('advanced', form.Flag, 'force_forwarding', _('Force Forwarding'),
+			_('Force forwarding even there is no public IP (please try STUN first)'))
+
 		s = m.section(form.GridSection, 'perm_rule', _('Service Access Control List'),
 			_('ACL specify which client addresses and ports can be mapped, IPv6 always allowed.'));
 		s.sortable = true;

--- a/luci-app-upnp/po/zh_Hans/upnp.po
+++ b/luci-app-upnp/po/zh_Hans/upnp.po
@@ -253,3 +253,9 @@ msgstr "上传速度"
 msgctxt "Use %s (%s = STUN)"
 msgid "Use %s"
 msgstr "使用 %s"
+
+msgid "Force Forwarding"
+msgstr "强制转发"
+
+msgid "Force forwarding even there is no public IP (please try STUN first)"
+msgstr "就算没有公网IP也进行转发（请先尝试STUN）"
