--- a/luci-app-subconverter/Makefile
+++ b/luci-app-subconverter/Makefile
@@ -12,7 +12,7 @@ define Package/$(PKG_NAME)
 	PKGARCH:=all
 	SUBMENU:=3. Applications
 	TITLE:=Subconverter
-	DEPENDS:=+luci-base +luci
+	DEPENDS:=+luci-base +luci +sub-web 
 endef
 
 define Package/$(PKG_NAME)/description
@@ -34,21 +34,18 @@ define Package/$(PKG_NAME)/prerm
     rm -rf /usr/subconverter
 endef
 
-
 define Package/$(PKG_NAME)/install
 	$(INSTALL_DIR) $(1)/usr/lib/lua/luci/controller
 	$(INSTALL_DIR) $(1)/usr/lib/lua/luci/view/subconverter
 	$(INSTALL_DIR) $(1)/etc/init.d
 
-	$(CP) -r $(TOPDIR)/package/$(PKG_NAME)/* $(PKG_BUILD_DIR)/
-
-	$(INSTALL_DATA) $(PKG_BUILD_DIR)/luasrc/controller/subconverter.lua $(1)/usr/lib/lua/luci/controller/
-	$(INSTALL_DATA) $(PKG_BUILD_DIR)/luasrc/view/subconverter/subconverter.htm $(1)/usr/lib/lua/luci/view/subconverter
-	$(INSTALL_DATA) $(PKG_BUILD_DIR)/luasrc/view/subconverter/prefini.htm $(1)/usr/lib/lua/luci/view/subconverter
+	$(INSTALL_DATA) ./luasrc/controller/subconverter.lua $(1)/usr/lib/lua/luci/controller/
+	$(INSTALL_DATA) ./luasrc/view/subconverter/subconverter.htm $(1)/usr/lib/lua/luci/view/subconverter
+	$(INSTALL_DATA) ./luasrc/view/subconverter/prefini.htm $(1)/usr/lib/lua/luci/view/subconverter
 	
-	$(CP) -r $(PKG_BUILD_DIR)/root/usr/subconverter $(1)/usr/
+	$(CP) -r ./root/usr/subconverter $(1)/usr/
 	
-	$(INSTALL_BIN) $(PKG_BUILD_DIR)/root/etc/init.d/subconverter $(1)/etc/init.d/
+	$(INSTALL_BIN) ./root/etc/init.d/subconverter $(1)/etc/init.d/
 endef
 
 $(eval $(call BuildPackage,$(PKG_NAME)))
