--- a/luci-app-zerotier/root/etc/zerotier.start
+++ b/luci-app-zerotier/root/etc/zerotier.start
@@ -18,6 +18,7 @@ rm -f "$nft_incdir/input/zerotier.nft" "$nft_incdir/forward/zerotier.nft" "$$nft
 
 nat_enable="$(uci get zerotier.sample_config.nat)"
 [ "$nat_enable" -eq "1" ] && {
+[ -n "$(command -v nft)" ] && {
 	[ -d "$nft_incdir/input" ] || mkdir -p "$nft_incdir/input"
 	[ -d "$nft_incdir/forward" ] || mkdir -p "$nft_incdir/forward"
 	[ -d "$nft_incdir/srcnat" ] || mkdir -p "$nft_incdir/srcnat"
@@ -34,4 +35,17 @@ nat_enable="$(uci get zerotier.sample_config.nat)"
 	uci -q set firewall.@defaults[0].auto_includes="1"
 	uci -q commit firewall
 	fw4 reload
+} || {
+	zt0="$(ifconfig | grep 'zt' | awk '{print $1}')"
+	echo "${zt0}" > "/tmp/zt.nif"
+	for i in ${zt0}
+	do
+		ip_segment=""
+		iptables -I FORWARD -i "$i" -j ACCEPT
+		iptables -I FORWARD -o "$i" -j ACCEPT
+		iptables -t nat -I POSTROUTING -o "$i" -j MASQUERADE
+		ip_segment="$(ip route | grep "dev $i proto kernel" | awk '{print $1}')"
+		iptables -t nat -I POSTROUTING -s "${ip_segment}" -j MASQUERADE
+	done
+}
 }

--- a/luci-app-zerotier/root/etc/init.d/zerotier
+++ b/luci-app-zerotier/root/etc/init.d/zerotier
@@ -94,10 +94,25 @@ start_service() {
 stop_instance() {
 	rm -f /tmp/zero.log
 	local cfg="$1"
-
+	[ -n "$(command -v nft)" ] && {
 	rm -f "/usr/share/nftables.d/chain-pre/forward/zerotier.nft" "/usr/share/nftables.d/chain-pre/srcnat/zerotier.nft"
 	fw4 reload
 	echo "zt interface rules removed!" > /tmp/zero.log 2>&1 &
+	} || {
+	zt0="$(ifconfig | grep 'zt' | awk '{print $1}')"
+	[ -z "${zt0}" ] && zt0="$(cat "/tmp/zt.nif")"
+
+	for i in ${zt0}
+	do
+	ip_segment=""
+	iptables -D FORWARD -i "$i" -j ACCEPT 2>/dev/null
+	iptables -D FORWARD -o "$i" -j ACCEPT 2>/dev/null
+	iptables -t nat -D POSTROUTING -o "$i" -j MASQUERADE 2>/dev/null
+	ip_segment="$(ip route | grep "dev $i proto" | awk '{print $1}')"
+	iptables -t nat -D POSTROUTING -s "${ip_segment}" -j MASQUERADE 2>/dev/null
+	echo "zt interface $i is stopped!"
+	done
+	}
 
 	# Remove existing link or folder
 	rm -f $CONFIG_PATH/networks.d/*.conf
