diff --git a/libfuse/lib/fmt/format.h b/libfuse/lib/fmt/format.h
index 7c607dbd..c08d6ca5 100644
--- a/libfuse/lib/fmt/format.h
+++ b/libfuse/lib/fmt/format.h
@@ -1604,10 +1604,10 @@ auto snprintf_float(T value, int precision, float_specs specs,
     abort_fuzzing_if(precision > 100000);
     // Suppress the warning about a nonliteral format string.
     // Cannot use auto because of a bug in MinGW (#1532).
-    int (*snprintf_ptr)(char*, size_t, const char*, ...) = FMT_SNPRINTF;
+    //int (*snprintf_ptr)(char*, size_t, const char*, ...) = FMT_SNPRINTF;
     int result = precision >= 0
-                     ? snprintf_ptr(begin, capacity, format, precision, value)
-                     : snprintf_ptr(begin, capacity, format, value);
+                     ? snprintf(begin, capacity, format, precision, value)
+                     : snprintf(begin, capacity, format, value);
     if (result < 0) {
       // The buffer will grow exponentially.
       buf.try_reserve(buf.capacity() + 1);
